### PR TITLE
dbaas: support for users

### PR DESCRIPTION
**Pull request by tgrondier**
# Description

For services that allow user creation, we are adding new resources that correspond to a user.

## Checklist
(For exoscale contributors)

* [ ] Changelog updated (under *Unreleased* block)
* [x] Acceptance tests OK
* [x] For a new resource, datasource or new attributes: acceptance test added/updated

## Testing

Local tests + acceptance tests

### CODE REVIEW

**Comment by kobajagi** [comment]
Also fixes https://github.com/exoscale/terraform-provider-exoscale/issues/326

**Comment by sauterp** [comment]
One [test](https://github.com/exoscale/terraform-provider-exoscale/actions/runs/12396342450/job/34604650054#step:6:444) seems to be failing. Probably unrelated to this PR but it's DBaaS related.  Just to be sure.

File: pkg/resources/database/resource_kafka_test.go [hunk 1]
```diff
@@ -158,7 +245,7 @@ func CheckExistsKafka(name string, data *TemplateModelKafka) error {
 
 	res, err := client.GetDbaasServiceKafkaWithResponse(ctx, oapi.DbaasServiceName(name))
 	if err != nil {
-		return err
+		return fmt.Errorf("aww")
```
Comment (sauterp) [inline pkg/resources/database/resource_kafka_test.go]:
```suggestion
		return err
```
leftover from debugging? :)

File: pkg/resources/database/resource_user.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"unable to change exoscale client zone",
+			err.Error(),
+		)
+		return
+	}
+
+	data.WaitForService(ctx, client, &resp.Diagnostics)
+	data.CreateResource(ctx, client, &diags)
```
Comment (sauterp) [inline pkg/resources/database/resource_user.go]: Is it possible you forgot to check the errors these functions return?

**Review by sauterp — APPROVED** [review]
LGTM, it's cool you managed to avoid some boilerplate through generics.

**Review by pierre-emmanuelJ — APPROVED** [review]

**Comment by tgrondier** [comment]
> One [test](https://github.com/exoscale/terraform-provider-exoscale/actions/runs/12396342450/job/34604650054#step:6:444) seems to be failing. Probably unrelated to this PR but it's DBaaS related. Just to be sure.

Yes I can't reproduce the issue, seems to have been transient.

**Comment by tgrondier** [comment]
> One [test](https://github.com/exoscale/terraform-provider-exoscale/actions/runs/12396342450/job/34604650054#step:6:444) seems to be failing. Probably unrelated to this PR but it's DBaaS related. Just to be sure.

I tested it a bit more on my side: it seems when a mysql dbaas goes from `rebuilding` to `running`, the user list returns only the default user `avnadmin` and not those created while the service was rebuilding.

Unsure how to handle that edge case in the tests

EDIT: Updated to wait for MySQL DBAAS to be running before allowing user updates
